plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/linked-product-control.tsx

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import ProductControl from '@woocommerce/editor-components/product-control';
+import { SelectedOption } from '@woocommerce/block-hocs';
+import { useState, useMemo } from '@wordpress/element';
+import { WooCommerceBlockLocation } from '@woocommerce/blocks/product-template/utils';
+import {
+	PanelBody,
+	PanelRow,
+	Button,
+	Flex,
+	FlexItem,
+	Dropdown,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalText as Text,
+	Spinner,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { useGetProduct } from '../../utils';
+import {
+	ProductCollectionQuery,
+	ProductCollectionSetAttributes,
+} from '../../types';
+
+const LinkedProductControl = ( {
+	query,
+	setAttributes,
+	location,
+	usesReference,
+}: {
+	query: ProductCollectionQuery;
+	setAttributes: ProductCollectionSetAttributes;
+	location: WooCommerceBlockLocation;
+	usesReference: string[] | undefined;
+} ) => {
+	const [ isDropdownOpen, setIsDropdownOpen ] = useState< boolean >( false );
+	const { product, isLoading } = useGetProduct( query.productReference );
+
+	const isShowLinkedProductControl = useMemo( () => {
```

Comment:
```suggestion
	const showLinkedProductControl = useMemo( () => {
```

I think it would be easier to read without `is`